apps/api/src/common/logger.service.ts

```diff
@@ -30,10 +30,19 @@ export class AppLogger implements LoggerService {
 
   error(e: unknown, context?: string) {
     const error = e as Error & { response?: Error };
-    let message = error.message;
+    let message: string | object = error.message;
+
+    if (typeof e === 'string') {
+      message = e;
+    }
 
     if (axios.isAxiosError(e)) {
-      message = (e as AxiosError).response?.data;
+      const { response, config } = e;
+      message = {
+        url: config.url,
+        method: config.method,
+        ...(response?.data ? { data: response.data } : {}),
```

Comment:
what if data contains any sensitive info or PI?

Reply:
This response would be what the external API is returning as error data also in this implementation the logger isn't responsible for filtering

We've added a bit of filtering in the error filter to prevent logging of the submission payload
We should also be diligent in not explicitly logging PII through info/warn methods